### DISCUSSION TITLE
504 gateway error

### DISCUSSION THREAD

**Discussion by iplayradionetworks** [magicbug/Cloudlog#2049] [General]
Hi . 

 I am experiencing an issue when I try to import from either an .adi file or from lotW . 

It takes time as the records are a lot ... Approx 20000+ qsos but most times I doesn't finish to the end and I get a 504 gateway error .

I have also increased the php time-out via php settings but still same issue.

Any ideas where to look or what I should be checking on my server?

Best 73s
Jimmy SY1DAL

**Comment by phl0** [comment]
Think I should answer here. For me it looks like an instance in the middle cannot talk to the backend/Cloudlog. The error is most likely not Cloudlog related ...

**Reply by iplayradionetworks** [reply]
Hm... Ok .

I just don't know where to look .

**Reply by phl0** [reply]
The logs of the apache server would be my first place to look.

**Reply by iplayradionetworks** [reply]
I will check .

Thanks .

**Comment by iplayradionetworks** [comment]
I tried again today and this is what I get ...

![20230404_122320.jpg](https://user-images.githubusercontent.com/40426604/229748536-c35a8e55-705d-41d5-87be-952240b09f46.jpg)

**Reply by phl0** [reply]
Okay this is an error coming from apache which happens if the backend does not answer in time. Is the apache configured to work as (reverse) proxy?
Maybe connectiontimeout is an option to tune. See https://httpd.apache.org/docs/2.4/mod/mod_proxy.html#workers

**Reply by iplayradionetworks** [reply]
Ok perhaps I must change connection timeout for apache.

I don't have apache as reverse proxy tho. Not needed . 

Will try the timeout of apache. I guess it will be easy to do

**Reply by iplayradionetworks** [reply]
Btw I see on top left corner "developer mode ". Why is that shown ?

Also any idea on how to delete duplicates from logbook ?

**Reply by AndreasK79** [reply]
Dev mode: https://github.com/magicbug/Cloudlog/wiki/Why-does-my-Cloudlog-install-show-Developer-Mode%3F

Duplicates can be found in the Duplicate QSOs search, which again can be found via the search button on top.

**Reply by iplayradionetworks** [reply]
Understood. 

Thanks 👍👍👍

**Reply by AndreasK79** [reply]
If you have lots of duplicates, it's quite time consuming to find and delete there. It can be done via the database, if you are comfortable running sql expressions. Let me know, and I can write a query that removes them.

**Reply by iplayradionetworks** [reply]
No it's ok. 

I don't have huge amount. 

Thanks 👍

**Reply by iplayradionetworks** [reply]
I have an issue .

My total qsos are 25440. 

When I import directly the adi file only 25291 are imported. 

What should I be checking ? Why 149 qsos are not imported?

What should I be looking in my adi file ? I have downloaded the Adif master application so I can edit and view my adi file . But what should I check ?

**Reply by phl0** [reply]
For a first try I would split the ADIF file in half an import the two. This way we can see if there are QSOs with issue or an issue with a max number of QSOs to import (which I don't believe).

**Reply by iplayradionetworks** [reply]
I shall try to split in half .

And import the first 12720 qsos .

Despite the fact that even after increasing the apache connection timeout I still get 504 error after import ends ... So I am not sure if the import completes 100%. There should be an indicator status bar during the process of import I suppose.... 

Will reply here with my feedback.

**Reply by iplayradionetworks** [reply]
After importing the first 10000 qsos from a new file, I still got 504 error . Anyways, only 9852 were imported.

Suggestions/ ideas ?

**Reply by phl0** [reply]
504 is a timeout error. For me it looks like the import is incomplete because of the timeout...

**Reply by iplayradionetworks** [reply]
I am sure that if I import 5000 qsos instead , import will be completed properly.... But can't be sure 💯 tho . 

How much more should I increase the timeout ?

**Reply by phl0** [reply]
tbh: I have no idea.

**Reply by iplayradionetworks** [reply]
So I guess there is no solution to my problem ...

**Reply by phl0** [reply]
I am pretty sure there is. But it is hard to debug from remote ... :(

**Reply by iplayradionetworks** [reply]
What if I try to send you my qsl file ? And you can see if it does the same to you ?

**Reply by phl0** [reply]
I can do that. Email on qrz.com. callsign is DF2ET.

**Reply by iplayradionetworks** [reply]
Ok will do that :) thanks

**Reply by iplayradionetworks** [reply]
Okies. 

Email sent ... Let me know once you've got it 😉

**Reply by phl0** [reply]
It did quite take some time but 25,305 QSOs got imported on my test system without any timeout or error. However some dupes have been skipped.

**Reply by phl0** [reply]
The output of the upload page showed 149 dupe entries that were skipped.

**Reply by iplayradionetworks** [reply]
Omg ... Thank you for this great help. 

So eventually I have indeed 149 dupes. They are not missing qsos .

That is good . I have to figure out in Adif master how to find dupes and then simply delete them before importing them into cloudlog .

One time when I tried to import even with 504 error I had imported 25291 total. So I guess it's correct. But I don't get to see the message afterwards the import regarding the dupes . 

I must find way to fix this 504 but don't know what else to do .

I have increased all connection timeout parameters but still same issue . 

Perhaps if I change php version for specific subdomain of where the app is installed might fix it ? Or if I change some php parameter that could be related ?

**Reply by iplayradionetworks** [reply]
Problem solved indeed in regards to qsos. 25440 are not correct. 25347 are correct. 

So I must remove dupes. 

Thanks again my friend.

Only problem left is regarding 504....

**Reply by GregTroar** [reply]
Hello,

I am having the same issue and tried to increase all timeout in php.ini and nginx but impossible to get rid of this 504 gateway timeout.
End of the day my QSOs are NOT imported so I cannot upload my adif which containers "only" 8,000 QSOs.

Any idea where to look, I checked the log of nginx and nothing wrong.

If I cannot import my log then I cannot really try cloudlog I am afraid.

Thanks

**Reply by AndreasK79** [reply]
Hi, divide your adif file. Either manually, or with a program designed for doing that. You can find one here: http://sp7dqr.pl/en/converters.php It is called Adif divider.

**Reply by GregTroar** [reply]
Ok got it working now !
Needed to change in the php.ini the max_post, it is working fine now.
Thanks
